pkg/mount/block/blockdev.go

```diff
@@ -0,0 +1,452 @@
+func (b *BlockDev) BlockSize() (int, error) {
+	f, err := os.Open(b.DevicePath())
+	if err != nil {
+		return 0, err
+	}
+	defer f.Close()
+	return unix.IoctlGetInt(int(f.Fd()), unix.BLKSSZGET)
+}
+
+// KernelBlockSize returns the soft block size used inside the kernel (BLKBSZGET).
+func (b *BlockDev) KernelBlockSize() (int, error) {
+	f, err := os.Open(b.DevicePath())
+	if err != nil {
+		return 0, err
+	}
+	defer f.Close()
+	return unix.IoctlGetInt(int(f.Fd()), unix.BLKBSZGET)
+}
+
+func ioctlGetUint64(fd int, req uint) (uint64, error) {
+	var value uint64
+	_, _, err := unix.Syscall(unix.SYS_IOCTL, uintptr(fd), uintptr(req), uintptr(unsafe.Pointer(&value)))
+	if err != 0 {
+		return 0, err
+	}
+	return value, nil
+}
+
+// Size returns the size in bytes.
+func (b *BlockDev) Size() (uint64, error) {
+	f, err := os.Open(b.DevicePath())
+	if err != nil {
+		return 0, err
+	}
+	defer f.Close()
+	return ioctlGetUint64(int(f.Fd()), unix.BLKGETSIZE64)
+}
+
+// ReadPartitionTable prompts the kernel to re-read the partition table on this block device.
+func (b *BlockDev) ReadPartitionTable() error {
+	f, err := os.OpenFile(b.DevicePath(), os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	return unix.IoctlSetInt(int(f.Fd()), unix.BLKRRPART, 0)
+}
+
+// SystemPartitionGUID is the GUID of EFI system partitions
+// EFI System partitions have GUID C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+var SystemPartitionGUID = gpt.Guid([...]byte{
+	0x28, 0x73, 0x2a, 0xc1,
+	0x1f, 0xf8,
+	0xd2, 0x11,
+	0xba, 0x4b,
+	0x00, 0xa0, 0xc9, 0x3e, 0xc9, 0x3b,
+})
+
+// Device makes sure the block device exists and returns a handle to it.
+//
+// maybeDevpath can be path like /dev/sda1, /sys/class/block/sda1 or just sda1.
+// We will just use the last component.
+func Device(maybeDevpath string) (*BlockDev, error) {
+	devname := filepath.Base(maybeDevpath)
+	if _, err := os.Stat(filepath.Join("/sys/class/block", devname)); err != nil {
+		return nil, err
+	}
+
+	devpath := filepath.Join("/dev/", devname)
+	if uuid, err := getUUID(devpath); err == nil {
+		return &BlockDev{Name: devname, FsUUID: uuid}, nil
+	}
+	return &BlockDev{Name: devname}, nil
+}
+
+// GetBlockDevices iterates over /sys/class/block entries and returns a list of
+// BlockDev objects, or an error if any
+func GetBlockDevices() (BlockDevices, error) {
+	var blockdevs []*BlockDev
+	var devnames []string
+
+	root := "/sys/class/block"
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		rel, err := filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		if rel == "." {
+			return nil
+		}
+		devnames = append(devnames, rel)
+		dev, err := Device(rel)
+		if err != nil {
+			return err
+		}
+		blockdevs = append(blockdevs, dev)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return blockdevs, nil
+}
+
+func getUUID(devpath string) (string, error) {
```

Comment:
Maybe rename to `getFsUUID` to be in sync with the field of `BlockDev` and because later we could maybe add functionality to discover the partition UUID and thus avoid confusion.

Reply:
sounds good. done.